_posts/2024-01-31-cockpit-310.md

```diff
@@ -0,0 +1,69 @@
+---
+title: Cockpit 310
+author: mvo
+date: '2024-01-31'
+tags: cockpit, machines
+slug: cockpit-310
+category: release
+summary: ''
+---
+
+Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).
+We release regularly.
+
+Here are the release notes from Cockpit 310 and cockpit-machines 306:
+
+## Storage: More support for btrfs
+
+Cockpit now understands btrfs filesystems even better. It can create new btrfs
+filesystems and new subvolumes.
+
+## Storage: improved support for swap
+
+Cockpit can now format block devices as swap and will maintain a fstab entry for them.
+
+## Machines: Action to Replace SPICE devices
+
+There are some VM host machines which don't support [SPICE
+features](https://www.spice-space.org/spice-user-manual.html). In
+particular, Red Hat Enterprise Linux 9 does not support SPICE at all,
+and on other operating systems the SPICE support packages may not be
+installed.
+
+The VM action menu got a "Replace SPICE support" action which prepares
+a VM so that upgrading/changing the host or migrating to a host
+without SPICE support will work without failing to start the VM
+afterwards. In particular, it converts SPICE graphics to VNC, "qxl"
+video to "vga", and drops all other spice devices which don't have a
+replacement (audio and USB passthrough).
+
+![screenshot of action to replace spice devices]({{ site.baseurl }}/images/310-machines-action-to-replace-spice-devices.png)
+
+If the host already doesn't support SPICE (any more), but the
+currently shown VM still uses SPICE, Cockpit shows a notification to
+explain how to fix it, which will lead to the same dialog:
```

Comment:
```suggestion
When a host does not have SPICE support, but the VM is configured for SPICE, then Cockpit will show an alert to explain the issue and provide a way to run the "replace SPICE" action. This may occur if a machine was migrated without replacing SPICE or if the host system was upgraded to a new version of a distribution that no longer supports SPICE.
```